subsys/bluetooth/mesh/op_agg.c

```diff
@@ -59,19 +59,19 @@ int bt_mesh_op_agg_send(struct bt_mesh_model *model,
 	/* Model responded so mark this message as acknowledged */
 	agg_ctx.ack = true;
 
-	if (IS_ENABLED(CONFIG_BT_MESH_OP_AGG_CLI)) {
-		/* Store source address so that Opcodes Aggregator Client can
-		 * match response with source model
-		 */
-		uint16_t src = bt_mesh_model_elem(model)->addr;
-
-		if (net_buf_simple_tailroom(&srcs) < 2) {
-			return -ENOMEM;
-		}
-
-		net_buf_simple_add_le16(&srcs, src);
+#if defined(CONFIG_BT_MESH_OP_AGG_CLI)
```

Comment:
Are you sure this is the best fix? Static-scope symbols and functions should get stripped out by the linker if there are no users. I.e. it should be possible to instead remove the `#if` guard from line 21. FWIW, I'm not 100% sure this is possible for variables, but at least for static functions it works and is a good way let the code keep using `IS_ENABLED()`.

Reply:
I believe I tried this initially and saw some leftovers from `NET_BUF_SIMPLE_DEFINE_STATIC` in .map file, but I checked this again and it seems working. I definitely messed up something. Should be better now.